guest-verona-rt/arch/x86_64/hv_sev.cc

```diff
@@ -409,20 +409,32 @@ namespace monza
       {
         kabort();
       }
-      if (entry_address < last_entry_end_address)
-      {
-        kabort();
-      }
       snmalloc::address_t current_entry_end_address =
         entry_address + entry_size;
-      // last_entry_end_address not used further this iteration.
-      last_entry_end_address = current_entry_end_address;
       if (
         current_entry_end_address < entry_address ||
         current_entry_end_address > (static_cast<uint64_t>(1) << 48))
       {
         kabort();
       }
+      // The first entry does not have to start at the heap start, but needs to
+      // include it
+      if (first_entry)
+      {
+        // last_entry_end_address is the start address of the heap here.
+        if (
+          last_entry_end_address < entry_address ||
+          last_entry_end_address >= current_entry_end_address)
```

Comment:
I think it would be clearer to use `heap_start`, i.e. then the comment is not required:
```suggestion
          snmalloc::address_cast(heap_start) < entry_address ||
          snmalloc::address_cast(heap_start) >= current_entry_end_address)
```

Reply:
If you changed:
```patch
< snmalloc::address_t last_entry_end_address = snmalloc::address_cast(heap_start);
> snmalloc::address_t last_entry_end_address = 0;
```
Then you can leave the original check the same, and add a new check for the `first_entry`.  Not needing the `else`.  I think that would be easier to follow.